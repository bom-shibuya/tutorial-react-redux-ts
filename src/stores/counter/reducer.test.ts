import * as snapshotDiff from 'snapshot-diff'
import { counter, initialState, minusNumber, plusNumber } from '.'

describe('Counter reducer test', () => {
  test('shold return initial state', () => {
    expect(counter(initialState, {} as any)).toBe(initialState)
  })

  test('shold return plus state', () => {
    expect(counter(10, plusNumber(5))).toBe(15)
  })

  test('shold return minus state', () => {
    expect(counter(10, minusNumber(5))).toBe(5)
  })
})

describe('Counter reducer snapshot', () => {
  test('shold return initial state', () => {
    expect(
      snapshotDiff(initialState, counter(initialState, {} as any))
    ).toMatchSnapshot()
  })
  test('shold return plus state', () => {
    expect(
      snapshotDiff(
        { counter: initialState },
        { counter: counter(initialState, plusNumber(5)) }
      )
    ).toMatchSnapshot()
  })
  test('shold return minus state', () => {
    expect(
      snapshotDiff(initialState, counter(initialState, minusNumber(5)))
    ).toMatchSnapshot()
  })
})
