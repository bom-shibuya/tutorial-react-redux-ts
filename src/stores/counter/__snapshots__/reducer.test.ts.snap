// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter reducer snapshot shold return initial state 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`Counter reducer snapshot shold return minus state 1`] = `
"Snapshot Diff:
null"
`;

exports[`Counter reducer snapshot shold return plus state 1`] = `
"Snapshot Diff:
- First value
+ Second value

  Object {
-   \\"counter\\": 0,
+   \\"counter\\": 5,
  }"
`;
